File: backend/src/main/java/com/librarymanagement/service/BookService.java
package com.librarymanagement.service;

import com.librarymanagement.dto.BookDTO;
import java.util.List;

public interface BookService {
    BookDTO createBook(BookDTO bookDTO);
    List<BookDTO> getAllBooks();
    BookDTO getBookById(Long id);
    BookDTO updateBook(Long id, BookDTO bookDTO);
    void deleteBook(Long id);
    List<BookDTO> searchBooks(String keyword);
    List<BookDTO> getBooksByCategory(String category);
}

File: frontend/src/services/bookService.js
import api from '../api';

const bookService = {
  getAllBooks: async () => {
    const response = await api.get('/books');
    return response.data;
  },

  searchBooks: async (keyword) => {
    const response = await api.get(`/books/search`, {
      params: { keyword }
    });
    return response.data;
  },

  getBooksByCategory: async (category) => {
    const response = await api.get(`/books/category/${encodeURIComponent(category)}`);
    return response.data;
  },

  createBook: async (bookData) => {
    const response = await api.post('/books', bookData);
    return response.data;
  },

  updateBook: async (id, bookData) => {
    const response = await api.put(`/books/${id}`, bookData);
    return response.data;
  },

  deleteBook: async (id) => {
    const response = await api.delete(`/books/${id}`);
    return response.data;
  }
};

export default bookService;
